src/registry/docset.cpp

```diff
@@ -241,11 +241,10 @@ const QMap<QString, QString> &Docset::symbols(const QString &symbolType) const
     return m_symbols[symbolType];
 }
 
-QList<SearchResult> Docset::search(const QString &query) const
+QList<SearchResult> Docset::search(const SearchQuery &searchQuery) const
```

Comment:
The approach here is wrong, and this change doesn't make it better. `Docset::search()` doesn't need to know anything about keywords. We should move the check below to `DocsetRegistry`.